app/controllers/reports_controller.rb

```diff
@@ -1,12 +1,18 @@
 class ReportsController < ServiceController
+  def index
+    raise ActiveRecord::RecordNotFound, 'Reports not found' unless current_user.can_access_reporting_dashboard?
+  end
+
   def show
+    raise ActiveRecord::RecordNotFound, 'Report not found' unless current_user.reports.include?(params[:id])
+
     case params[:id]
     when /workload_report/
       @report = Reporting::WorkloadReport.new
     when /processed_report/
       @report = Reporting::ProcessedReport.new
-    else
-      raise ActiveRecord::RecordNotFound, 'Report not found'
+    when /caseworker_report/
+      @report = Reporting::CaseworkerReport.new
     end
```

Comment:
Definitely no need for the `else` ?

Reply:
You are right. The controller might not support all user_reports.... Thanks!